lib/rubocop/cop/metrics/block_length.rb

```diff
@@ -33,23 +33,29 @@ module Metrics
       # NOTE: This cop does not apply for `Struct` definitions.
       class BlockLength < Base
         include CodeLength
+        include IgnoredMethods
+
+        ignored_methods deprecated_key: 'ExcludedMethods'
 
         LABEL = 'Block'
 
         def on_block(node)
-          return if excluded_method?(node)
+          return if ignored_method?(node.method_name)
+          return if method_receiver_excluded?(node)
           return if node.class_constructor? || node.struct_constructor?
 
           check_code_length(node)
         end
 
         private
 
-        def excluded_method?(node)
+        def method_receiver_excluded?(node)
           node_receiver = node.receiver&.source&.gsub(/\s+/, '')
           node_method = String(node.method_name)
 
-          excluded_methods.any? do |config|
+          ignored_methods.any? do |config|
+            next unless config.is_a?(String)
+
             receiver, method = config.split('.')
```

Comment:
`Metrics/BlockLength` has a special case for `IgnoredMethods`, where it allows for values to be fully qualified methods, and it uses this to test receivers. I kept this logic of course, but this is now an exception from how all the other `IgnoredMethods` cops work.